contracts/polygon/connectors/gelato/aave-services/protection/main.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+/**
+ * @title Aave Protection.
+ * @dev Protect DSA against Liquidation risk on Aave with Gelato.
+ */
+
+
+import {IERC20} from "./interface.sol";
+import {Helpers} from "./helpers.sol";
+
+abstract contract GAaveProtectionResolver is Helpers {
+    /// @dev Function for submitting a protection task
+    /// @param _wantedHealthFactor targeted health after protection.
+    /// @param _minimumHealthFactor trigger protection when current health
+    /// factor is below _minimumHealthFactor.
+    /// @param _isPermanent boolean to set a protection as permanent
+    function submitProtection(
+        uint256 _wantedHealthFactor,
+        uint256 _minimumHealthFactor,
+        bool _isPermanent
+    ) external payable {
+        _submitProtection(
+            _wantedHealthFactor,
+            _minimumHealthFactor,
+            _isPermanent
+        );
+    }
+
+    /// @dev Function for modifying a protection task
+    /// @param _wantedHealthFactor targeted health after protection.
+    /// @param _minimumHealthFactor trigger protection when current health
+    /// factor is below _minimumHealthFactor.
+    /// @param _isPermanent boolean to set a protection as permanent
+    function updateProtection(
+        uint256 _wantedHealthFactor,
+        uint256 _minimumHealthFactor,
+        bool _isPermanent
+    ) external payable {
+        _updateProtection(
+            _wantedHealthFactor,
+            _minimumHealthFactor,
+            _isPermanent
+        );
+    }
+
+    /// @dev Function for cancelling a protection task
+    function cancelProtection() external payable {
+        _cancelProtection();
+    }
+
+    /// @dev Function for cancelling and removing allowance
+    /// of aToken to _protectionAction
+    function cancelAndRevoke() external payable {
+        if (_dsaHasProtection()) _cancelProtection();
+        _revokeAllowance();
+    }
+
+    function _submitProtection(
```

Comment:
Can you move all the internal functions to `helpers.sol`? `main.sol` are mainly for functions that are going to be called by DSAs.

Reply:
Yes all internals function from ```main.sol``` has been moved to ```helpers.sol```.